c/cmake_modules/GoUtils.cmake

```diff
@@ -87,10 +87,10 @@ function(add_go_lib GO_MOD_DIR GO_LIBNAME)
   separate_arguments(GO_BUILD_FLAGS NATIVE_COMMAND
                      "${GO_BUILD_FLAGS} $<$<CONFIG:DEBUG>:-gcflags=\"-N -l\">")
 
-  # if we're building debug mode then change the default CGO_CFLAGS and CGO_CXXFLAGS from "-g O2" to "-g3"
-  set(GO_ENV_VARS
-      "CGO_ENABLED=1 $<$<CONFIG:DEBUG>:CGO_CFLAGS=-g3> $<$<CONFIG:DEBUG>:CGO_CXXFLAGS=-g3>"
-  )
+  # if we're building debug mode then change the default and CGO_CXXFLAGS from "-g O2" to "-g3"
+  # We can't add CGO_CFLAGS=-g3 because it will result in no longer inlining the ArrowArray* helper
+  # functions used internally within the cdata Go package, causing linker errors of missing symbols.
+  set(GO_ENV_VARS "CGO_ENABLED=1 $<$<CONFIG:DEBUG>:CGO_CXXFLAGS=-g3>")
```

Comment:
Hmm, whether something is inlined or not shouldn't affect whether the symbol can be found...Should this be fixed upstream?

Reply:
I'm mostly just assuming this is the case because the functions which are showing up as undefined symbols are all the inline funcs in helpers.h

I'm only able to reproduce the missing symbols when I add `CGO_CFLAGS=-g3` and build the driver shared object. I honestly don't even know what the upstream fix would be...

Reply:
https://stackoverflow.com/questions/62374711/c-inline-function-generates-undefined-symbols-error

It looks like they should all be `static inline`

Reply:
Makes sense, I'll futz with them tomorrow and see if that fixes the issue.

Reply:
w00t, it fixed it!